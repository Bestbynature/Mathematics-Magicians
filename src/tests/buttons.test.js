import { render } from '@testing-library/react';
import Buttons from '../components/buttons';
import updateDisplay from '../components/calculator';

describe('Buttons', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Buttons updateDisplay={(e) => updateDisplay(e)} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render 19 buttons', () => {
    const { getAllByRole } = render(<Buttons updateDisplay={(e) => updateDisplay(e)} />);
    expect(getAllByRole('button')).toHaveLength(19);
  });

  it('should render 5 buttons with class operations', () => {
    const { container } = render(<Buttons updateDisplay={(e) => updateDisplay(e)} />);
    const operationsButtons = container.querySelectorAll('.operations');
    expect(operationsButtons).toHaveLength(5);
  });

  it('should render 1 button with class rowspan', () => {
    const { container } = render(<Buttons updateDisplay={(e) => updateDisplay(e)} />);
    const rowspanButton = container.querySelectorAll('.rowspan');
    expect(rowspanButton).toHaveLength(1);
  });
});
