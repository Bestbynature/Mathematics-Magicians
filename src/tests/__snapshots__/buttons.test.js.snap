// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render correctly 1`] = `
<DocumentFragment>
  <div
    class="buttons-cont"
  >
    <button
      type="button"
    >
      AC
    </button>
    <button
      type="button"
    >
      +/-
    </button>
    <button
      type="button"
    >
      %
    </button>
    <button
      class="operations"
      type="button"
    >
      รท
    </button>
    <button
      type="button"
    >
      7
    </button>
    <button
      type="button"
    >
      8
    </button>
    <button
      type="button"
    >
      9
    </button>
    <button
      class="operations"
      type="button"
    >
      ร
    </button>
    <button
      type="button"
    >
      4
    </button>
    <button
      type="button"
    >
      5
    </button>
    <button
      type="button"
    >
      6
    </button>
    <button
      class="operations"
      type="button"
    >
      -
    </button>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      class="operations"
      type="button"
    >
      +
    </button>
    <button
      class="rowspan"
      type="button"
    >
      0
    </button>
    <button
      type="button"
    >
      .
    </button>
    <button
      class="operations"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
